### PR TITLE
feat: ability to set background colors to entries

### LINKED ISSUE #26: Please add a HEX-BackgroundColor option for the TokenVault entries
It would be great if you could add a background-color option (hex code) for the TokenVault-Entries, since I have a lot of entries which I use. Specifying a background color would help me a lot to find the corresponding entry much faster?

I also don't think that this would be hard to implement?

### PR DESCRIPTION
**Pull request by johnny243**
- [x] set/clear color per entry
- [x] select from a list of predefined colors or type your color code manually

Peek 2021-05-25 00-58

closes #26

### CODE REVIEW

**Comment by mobitar** [comment]
Nice!

File: app/components/EditEntry.js [hunk 1]
```diff
@@ -57,22 +91,66 @@ export default class EditEntry extends React.Component {
   };
 
   render() {
-    const { id, entry } = this.state;
+    const { id, entry, showColorPicker: showColorPicker } = this.state;
+
+    const defaultBgColor = '#fff';
+    const defaultColorOptions = [
```
Comment (mobitar) [inline app/components/EditEntry.js:97]: This should be extracted outside of the render function I think. Maybe in its own file or at the top of this file.

File: app/components/EditEntry.js [hunk 2]
```diff
@@ -57,22 +91,66 @@ export default class EditEntry extends React.Component {
   };
 
   render() {
-    const { id, entry } = this.state;
+    const { id, entry, showColorPicker: showColorPicker } = this.state;
+
+    const defaultBgColor = '#fff';
```
Comment (mobitar) [inline app/components/EditEntry.js:96]: Should this be inserted into the array of options?

File: app/components/EditEntry.js [hunk 3]
```diff
@@ -57,22 +91,66 @@ export default class EditEntry extends React.Component {
   };
 
   render() {
-    const { id, entry } = this.state;
+    const { id, entry, showColorPicker: showColorPicker } = this.state;
```
Comment (mobitar) [inline app/components/EditEntry.js:94]: Would this work?

```suggestion
    const { id, entry, showColorPicker } = this.state;
```

**Review by mobitar — APPROVED** [review]

File: app/components/EditEntry.js (hunk 2 above)
Comment (johnny243) [reply app/components/EditEntry.js]: This is just used to show that the entry doesn't have a color. Users can type this color manually if they'd like to use it, though.

**Pull request merged**